core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -190,7 +192,8 @@ private[kafka] object ConsumerStage {
       if (offsets.isEmpty)
         Future.successful(Done)
       else {
-        stages.head._2.commit(this)
+        implicit val ec = akka.dispatch.ExecutionContexts.global()
+        Future.sequence(stages.map(_._2.commit(this))).map(_ => Done)
```

Comment:
It can be done if `CommittableOffsetBatch` exposes the stages alongside the offsets, and once done the Committer doesn't have to be a `KafkaAsyncConsumerCommitterRef` anymore since the owned `ActorRef` is redundant.